AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -639,62 +646,93 @@ - (BOOL)isLayerBacked
   return _flags.layerBacked;
 }
 
-#pragma mark - Layout measurement calculation
+#pragma mark - Layout measurement and sizing
+
+- (ASSize)size
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  return _size;
+}
+
+- (void)setSize:(ASSize)size
+{
+  ASDN::MutexLocker l(__instanceLock__);
+  if (ASSizeEqualToSize(_size, size) == NO) {
+    _size = size;
+    [self invalidateCalculatedLayout];
```

Comment:
At the moment it does not need to hold the lock necessarily, but we should still hold the lock for the whole operation though. It can happen that the method will change that would need to have the lock for the whole operation.